server/sonar-server/src/test/java/org/sonar/server/component/ws/SuggestionsActionTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.component.ws;
+
+import java.util.Collections;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.sonar.api.config.MapSettings;
+import org.sonar.api.resources.Qualifiers;
+import org.sonar.api.utils.System2;
+import org.sonar.core.util.stream.Collectors;
+import org.sonar.db.DbTester;
+import org.sonar.db.component.ComponentDto;
+import org.sonar.db.organization.OrganizationDto;
+import org.sonar.db.organization.OrganizationTesting;
+import org.sonar.server.component.index.ComponentIndex;
+import org.sonar.server.component.index.ComponentIndexDefinition;
+import org.sonar.server.component.index.ComponentIndexer;
+import org.sonar.server.es.EsTester;
+import org.sonar.server.permission.index.PermissionIndexerTester;
+import org.sonar.server.tester.UserSessionRule;
+import org.sonarqube.ws.WsComponents.SuggestionsWsResponse;
+
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.api.security.DefaultGroups.ANYONE;
+import static org.sonar.db.component.ComponentTesting.newProjectDto;
+
+public class SuggestionsActionTest {
+
+  @Rule
+  public DbTester db = DbTester.create(System2.INSTANCE);
+
+  @Rule
+  public EsTester es = new EsTester(new ComponentIndexDefinition(new MapSettings()));
+
+  @Rule
+  public UserSessionRule userSessionRule = UserSessionRule.standalone();
+
+  private ComponentIndex index;
+  private ComponentIndexer indexer;
+  private SuggestionsAction action;
+  private OrganizationDto organization;
+  private PermissionIndexerTester authorizationIndexerTester = new PermissionIndexerTester(es);
+
+  @Before
+  public void setUp() {
+    index = new ComponentIndex(es.client(), userSessionRule);
+    indexer = new ComponentIndexer(db.getDbClient(), es.client());
+    action = new SuggestionsAction(db.getDbClient(), index);
+    organization = OrganizationTesting.newOrganizationDto();
+  }
+
+  @Test
+  public void exact_match_in_one_qualifier() {
+    ComponentDto dto = newProjectDto(organization, "project-uuid").setId(42l);
```

Comment:
for readability, `42L` is preferred over `42l`, which can look like 421 with some fonts.